hist/hist/src/TH1.cxx

```diff
@@ -8230,11 +8247,17 @@ Double_t TH1::KolmogorovTest(const TH1 *h2, Option_t *option) const
 
       // make nEXPT experiments (this should be a parameter)
       prb3 = 0;
+      if (afunc1) Copy(*h1Expt);
+      if (afunc2) h2->Copy(*h2Expt);
```

Comment:
```suggestion
```
Is this even necessary? The `h1Exp` is already created as a clone of `this`, and the `h2Exp` could also be created as a clone of `h2`.

By the way, when you create the clones, I would use a lambda to reduce code duplication and also use `std::unique_ptr`, replacing the block a few lines before with:
```c++
      auto cloneHist = [&](TH1 const *hist) {
         return std::unique_ptr<TH1>{static_cast<TH1 *>((gDirectory ? gDirectory : gROOT)->CloneObject(hist, false))};
      };
      std::unique_ptr<TH1> h1_cpy = cloneHist(this);
      std::unique_ptr<TH1> h1Expt = cloneHist(this);
      std::unique_ptr<TH1> h2Expt = cloneHist(this /*or maybe h2 here?*/);
```

Reply:
Thanks for the comment. I will simplify the code using just copy and avoid un-neeeded clones of histograms